owl2nl/src/main/java/org/aksw/owl2nl/Optimizer.java

```diff
@@ -60,12 +66,149 @@ else if ((list.get(i)).Key.equals("CC")) {
+                     //   loopSize=ccList.size();
+                   // }
+
+                for (int i=0;i<loopSize;i++){
+                        if (ccListPointer<ccList.size() && commaListPointer<commaList.size()){
+                            if (commaList.get(commaListPointer)<ccList.get(ccListPointer)){
+                                combinedCcCommaList.add(commaList.get(commaListPointer));
+                                commaListPointer++;
+                            }
+                            else{
+                                combinedCcCommaList.add(ccList.get(ccListPointer));
+                                ccListPointer++;
+                            }
+                        }
+                        else if (ccListPointer>=ccList.size()){
+                            combinedCcCommaList.add(commaList.get(commaListPointer));
+                            commaListPointer++;
+                        }
+                        else if (commaListPointer>=commaList.size()){
+                            combinedCcCommaList.add(ccList.get(ccListPointer));
+                            ccListPointer++;
+                        }
+                }
+                    /*check whether the verbs are same or not*/
+                for(int j=1;j<verbList.size();j++){
+                    if(!(verbList.get(0).Value.equals(verbList.get(j).Value))){
+                        //not same verb
+                        sameVerb=false;
+                        return text;
+
+                    }
+                    sameVerb=true;
+
+                }
+                /*join*/
+                if (sameVerb && verbList.size()==(1+combinedCcCommaList.size())){
+                    //int startAdding;
+                    //System.out.println("loopszie="+loopSize);
+                    for (int i = 0; i <= combinedCcCommaList.get(0); i++) {
+                        finalText.add(list.get(i).Value);
+                    }
+                    for(int j=1;j<loopSize+1;j++){
+                       // if (j==0) {
+
+                        //}
+                         if(j==loopSize){
+                            for (int i = Integer.parseInt(verbList.get(j).Key) + 1; i < list.size(); i++) {
+                                finalText.add(list.get(i).Value);
+                            }
+                         }
+                        else{
+                            for (int i = Integer.parseInt(verbList.get(j).Key) + 1; i <= combinedCcCommaList.get(j); i++) {
+                                finalText.add(list.get(i).Value);
+                            }
+                        }
+
+                    }
+
+
+                }
+
+                //}
+
+            }
+            else{
+               // return text;
+            }
+           // if (sameVerb) {
+                text = String.join(" ", finalText);
+                sameVerb=false;
+            //}
             System.out.println("After Optimize : " + text);
```

Comment:
Please change "After Optimise" to "Text After our changes"

Reply:
done

Reply:
done